scripts/lint-package-lock.js

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env node
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+const fs = require('fs');
+const path = require('path');
+
+/**
+ * 1. Checks if package-lock file is from npm8
+ * 2. Checks for any dependencies that incorrectly use CodeArtifact instead of npm.
+ * 3. Check for any @cloudscape-design dependencies
+ */
+const filename = path.resolve(__dirname, '..', 'package-lock.json');
+const packageLock = JSON.parse(fs.readFileSync(filename));
+
+if (packageLock.lockfileVersion !== 2) {
+  throw new Error('package-lock.json must have "lockfileVersion": 2');
+}
+
+function checkDependencies(dependencyName, dependencies) {
+  const dependency = dependencies[dependencyName];
+
+  if (dependencyName.includes('@cloudscape-design/')) {
+    throw Error('package-lock.json file contains @cloudscape-design dependency');
+  }
+
+  if (dependency && dependency.resolved && dependency.resolved.includes('codeartifact.us-west-2.amazonaws.com')) {
```

Comment:
How about removing this script completely (keep only `unlock-package-lock.js`)

if lockfile contains any invalid registries, the build will fail regardless